core/src/main/java/io/questdb/griffin/engine/functions/math/RoundDownDoubleFunctionFactory.java

```diff
@@ -0,0 +1,158 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2020 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.math;
+
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.griffin.engine.functions.DoubleFunction;
+import io.questdb.griffin.engine.functions.UnaryFunction;
+import io.questdb.griffin.engine.functions.constants.DoubleConstant;
+import io.questdb.std.Numbers;
+import io.questdb.std.NumericException;
+import io.questdb.std.ObjList;
+
+
+public class RoundDownDoubleFunctionFactory implements FunctionFactory {
+
+    @Override
+    public String getSignature() {
+        return "rounddown(DI)";
+    }
+
+    @Override
+    public Function newInstance(ObjList<Function> args, int position, CairoConfiguration configuration) {
+        Function precision = args.getQuick(1);
+        if (precision.isConstant()) {
+            int precisionValue = precision.getInt(null);
+            if (precisionValue != Numbers.INT_NaN && Math.abs(precisionValue) <= Numbers.pow10max) {
+                if (precisionValue >= 0) {
+                    return new FuncPosConst(position, args.getQuick(0), precisionValue);
+                }
+                return new FuncNegConst(position, args.getQuick(0), precisionValue);
```

Comment:
here is a suggestion: when we pass `precisionValue` as is to this constructor we will have to execute `-scale` for every row. If on other hand we pass `-previsionValue` the functions can be rewritten to take positive scale value and but interpret it as negative. Thus not execute `-` operator for every row.

Reply:
Will leage open for the moment - can only do once I convert other functions - round() and roundup() to the same model.